### ISSUE TITLE
AWS Lambda Destination

### ISSUE THREAD

**Issue by bluebeel** [serverless/serverless#7007]
# AWS Lambda Destination

## Description

[AWS Lambda Destinations](https://aws.amazon.com/fr/blogs/compute/introducing-aws-lambda-destinations/) for asynchronous invocations. This is a feature that provides visibility into Lambda function invocations and routes the execution results to AWS services, simplifying event-driven applications and reducing code complexity.

I have already simply tried to add a cloudformation resource: 

```
functions:
  hello:
    handler: src/hello.handler
    events:
      - sqs:
          batchSize: 1
          arn: ${arn}

resources: # CloudFormation template syntax
  Resources:
    HelloLambdaFunction:
      Type: AWS::Lambda::EventInvokeConfig
      Properties:
          FunctionName: "hello"
          Qualifier: "$LATEST"
          MaximumEventAgeInSeconds: 600
          MaximumRetryAttempts: 0
          DestinationConfig:
              OnSuccess:
                  Destination: ${arn}
              OnFailure:
                  Destination: ${arn}
```

But i got this error: `Update of resource type is not permitted. The new template modifies resource type of the following resources: [HelloLambdaFunction]`

Yours sincerely.

**Comment by medikoo** [comment]
@bluebeel as I see problem is, that you named your `AWS::Lambda::EventInvokeConfig` resource same as `AWS::Lambda::Function` resource is named. Ensure to name it differently.

Anyway, Framework will shortly support AWS Lambda Destination out of a box.

**Comment by bluebeel** [comment]
I had to change the resource name. Thanks it's working :)

**Comment by JoeNyland** [comment]
@medikoo It's great to hear that Serverless plans to provide support for Lambda destinations! I see the card [here](https://github.com/serverless/serverless/projects/7#card-29728045). Would you be able to give a rough estimate on how long it will take until work starts on this feature? Are we talking days, weeks, < 1 month, < 3 months, or > 3 months?

**Comment by vladgolubev** [comment]
Looks like CloudFormation support just landed: https://aws.amazon.com/about-aws/whats-new/2019/12/aws-cloudformation-updates-for-api-gateway-codepipeline-s3-iam-ecs-rds-es-lambda-and-more/

**Comment by danksky** [comment]
> @bluebeel as I see problem is, that you named your `AWS::Lambda::EventInvokeConfig` resource same as `AWS::Lambda::Function` resource is named. Ensure to name it differently.
> 
> Anyway, Framework will shortly support AWS Lambda Destination out of a box.

@medikoo  Where do you see this in his snippet?

**Comment by medikoo** [comment]
> @medikoo Where do you see this in his snippet?

` HelloLambdaFunction` it's the name that Serverless Framework will auto generate for configured `hello` functionn

**Comment by medikoo** [comment]
Will be addressed with #7261
